pkg/controller/provider/container/ovirt/model.go

```diff
@@ -837,7 +837,7 @@ func (r *VMAdapter) Event() []int {
 // List the collection.
 func (r *VMAdapter) List(ctx *Context) (itr fb.Iterator, err error) {
 	vmList := VMList{}
-	err = ctx.client.list("vms", &vmList, r.follow())
+	err = ctx.client.list("vms", &vmList, r.follow(), libweb.Param{Key: "all_content", Value: "true"})
```

Comment:
that's surprising, I guess you did this because you didn't see the tpm_enabled without it, right?
forklift gets snapshots and other things that are not returned by non-extended queries, did you check how we get them?

Reply:
yeah we use follow params. `?follow=snapshots....`
however here we need all content as this is the only way to get it into the vm. in ovirt it queries the vm devices and then indicate it for us.

Reply:
yes, I'm concerned about the performance penalty by this change
we have validations for watchdog and balloon devices - you say that the tpm device behaves differently?

Reply:
for watchdog we use follow. [here](https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendVmResource.java#L640) we can see the list of devices we need the use of querying the vm devices and using all-content.

Reply:
so how about adding the tpm with a follow? we could do something more clever then like querying tpm only for VMs with UEFI

Reply:
I meant in oVirt, we can still make change in RHV

Reply:
let's say we do (it means adding a follow link and a new backendTpmDevice +-), supposedly it will be an addition to the option of all_content, in order to not break back compatibility. but then, you will need a very late version of ovirt to support the follow part. i don't see it a reasonable solution.

Reply:
let's summarize here what we discussed offline:
adding an indication of TPM to oVirt/RHV is possible but complex
we can alternatively do that by the operating system - generate a warning only for Win 11 and 2022 that require TPM (it won't cover all cases, like Linux with TPM, but it should cover most of the cases with much lower effort)